cpp/src/arrow/ipc/feather.cc

```diff
@@ -444,6 +444,66 @@ class TableReader::TableReaderImpl {
     return Status::OK();
   }
 
+  Status Read(std::shared_ptr<Table>* out) {
+    std::vector<std::shared_ptr<Field>> fields;
+    std::vector<std::shared_ptr<Column>> columns;
+    for (int i = 0; i < num_columns(); ++i) {
+      std::shared_ptr<Column> column;
+      RETURN_NOT_OK(GetColumn(i, &column));
+      columns.push_back(column);
+      fields.push_back(column->field());
+    }
+    *out = Table::Make(schema(fields), columns);
+    return Status::OK();
+  }
+
+  Status Read(const std::vector<int>& indices, std::shared_ptr<Table>* out) {
+    std::vector<std::shared_ptr<Field>> fields;
+    std::vector<std::shared_ptr<Column>> columns;
+    for (int i = 0; i < num_columns(); ++i) {
+      bool found = false;
+      for (auto j : indices) {
+        if (i == j) {
+          found = true;
+          break;
+        }
+      }
+      if (!found) {
+        continue;
+      }
+      std::shared_ptr<Column> column;
+      RETURN_NOT_OK(GetColumn(i, &column));
```

Comment:
First, I implemented so but I stopped. Because it breaks backward compatibility for the original Python implementation. (The Python implementation ignores invalid indices.)
Can we accept the backward compatibility?